collector/remote_fx.go

```diff
@@ -0,0 +1,472 @@
+		),
+		InputFramesPerSecond: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gfx_input_frames_persec"),
+			"Number of sources frames provided as input to RemoteFX graphics per second.",
+			[]string{"session"},
+			nil,
+		),
+		OutputFramesPerSecond: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gfx_output_frames_persec"),
+			"Number of frames sent to the client per second.",
+			[]string{"session"},
+			nil,
+		),
+		SourceFramesPerSecond: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "gfx_source_frames_persec"),
+			"Number of frames composed by the source (DWM) per second.",
+			[]string{"session"},
+			nil,
+		),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *RemoteFxCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collectRemoteFXNetworkCount(ctx, ch); err != nil {
+		log.Error("failed collecting terminal services session count metrics:", desc, err)
+		return err
+	}
+	if desc, err := c.collectRemoteFXGraphicsCounters(ctx, ch); err != nil {
+		log.Error("failed collecting terminal services session count metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+type perflibRemoteFxNetwork struct {
+	Name                     string
+	BaseTCPRTT               float64 `perflib:"Base TCP RTT"`
+	BaseUDPRTT               float64 `perflib:"Base UDP RTT"`
+	CurrentTCPBandwidth      float64 `perflib:"Current TCP Bandwidth"`
+	CurrentTCPRTT            float64 `perflib:"Current TCP RTT"`
+	CurrentUDPBandwidth      float64 `perflib:"Current UDP Bandwidth"`
+	CurrentUDPRTT            float64 `perflib:"Current UDP RTT"`
+	FECRate                  float64 `perflib:"FEC Rate"`
+	LossRate                 float64 `perflib:"Loss Rate"`
+	RetransmissionRate       float64 `perflib:"Retransmission Rate"`
+	TCPReceivedRate          float64 `perflib:"TCP Received Rate"`
+	TCPSentRate              float64 `perflib:"TCP Sent Rate"`
+	TotalReceivedRate        float64 `perflib:"Total Received Rate"`
+	TotalSentRate            float64 `perflib:"Total Sent Rate"`
+	TotalReceivedBytes       float64 `perflib:"Total Received Bytes"`
+	TotalSentBytes           float64 `perflib:"Total Sent Bytes"`
+	UDPPacketsReceivedPersec float64 `perflib:"UDP Packets Received/sec"`
+	UDPPacketsSentPersec     float64 `perflib:"UDP Packets Sent/sec"`
+	UDPReceivedRate          float64 `perflib:"UDP Received Rate"`
+	UDPSentRate              float64 `perflib:"UDP Sent Rate"`
+}
+
+func (c *RemoteFxCollector) collectRemoteFXNetworkCount(ctx *ScrapeContext, ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	dst := make([]perflibRemoteFxNetwork, 0)
+	err := unmarshalObject(ctx.perfObjects["RemoteFX Network"], &dst)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, d := range dst {
+		ch <- prometheus.MustNewConstMetric(
+			c.BaseTCPRTT,
+			prometheus.GaugeValue,
+			float64(d.BaseTCPRTT),
```

Comment:
You can remove the `float64()` cast on all metrics as the perflib metrics are already defined as float64 in `perflibRemoteFxNetwork` and `perflibRemoteFxGraphics`.